### ISSUE TITLE
Workbench tabs retract on their own

### ISSUE THREAD

**Issue by DankestO** [Truly-Modular/Modular-Item-API#17]
**Describe the bug**

When I select a module, it automatically goes back to the previous tab

Curiously, it doesn't happen at all with swords if after creating the handle, I make the pommer and then the guard, if I make the guard first, it doesn't let me place the pommer.

E̶d̶i̶t̶:̶ ̶I̶t̶ ̶s̶e̶e̶m̶s̶ ̶t̶h̶a̶t̶ ̶i̶t̶ ̶h̶a̶p̶p̶e̶n̶s̶ ̶w̶i̶t̶h̶ ̶t̶h̶e̶ ̶M̶e̶k̶a̶n̶i̶s̶m̶ ̶m̶e̶t̶a̶l̶s̶,̶ ̶i̶t̶ ̶s̶e̶e̶m̶s̶ ̶t̶h̶a̶t̶ ̶i̶t̶ ̶d̶o̶e̶s̶n̶'̶t̶ ̶h̶a̶p̶p̶e̶n̶ ̶w̶i̶t̶h̶ ̶t̶h̶e̶ ̶r̶e̶s̶t̶ ̶o̶f̶ ̶t̶h̶e̶ ̶m̶e̶t̶a̶l̶s̶ ̶f̶r̶o̶m̶ ̶o̶t̶h̶e̶r̶ ̶m̶o̶d̶s̶ ̶t̶h̶a̶t̶ ̶I̶ ̶h̶a̶v̶e̶,̶ ̶h̶o̶w̶ ̶s̶t̶r̶a̶n̶g̶e̶.̶ ̶I̶f̶,̶ ̶f̶o̶r̶ ̶e̶x̶a̶m̶p̶l̶e̶,̶ ̶I̶ ̶p̶u̶t̶ ̶t̶h̶e̶ ̶h̶a̶n̶d̶l̶e̶ ̶o̶f̶ ̶a̶n̶ ̶a̶x̶e̶ ̶w̶i̶t̶h̶ ̶b̶r̶o̶n̶z̶e̶ ̶(̶M̶e̶k̶a̶n̶i̶s̶m̶)̶,̶ ̶t̶h̶e̶ ̶t̶a̶b̶s̶ ̶b̶e̶g̶i̶n̶ ̶t̶o̶ ̶b̶u̶g̶ ̶a̶s̶ ̶I̶ ̶m̶e̶n̶t̶i̶o̶n̶e̶d̶,̶ ̶i̶t̶ ̶s̶h̶o̶w̶s̶ ̶a̶s̶ ̶w̶o̶o̶d̶,̶ ̶b̶u̶t̶ ̶i̶t̶'̶s̶ ̶a̶c̶t̶u̶a̶l̶l̶y̶ ̶s̶u̶p̶p̶o̶s̶e̶d̶ ̶t̶o̶ ̶b̶e̶ ̶b̶r̶o̶n̶z̶e̶.̶

A̶ ̶n̶o̶t̶i̶c̶e̶ ̶a̶l̶s̶o̶ ̶a̶p̶p̶e̶a̶r̶s̶ ̶o̶n̶ ̶t̶h̶e̶ ̶d̶e̶d̶i̶c̶a̶t̶e̶d̶ ̶s̶e̶r̶v̶e̶r̶:̶
*̶*̶C̶o̶u̶l̶d̶ ̶n̶o̶t̶ ̶p̶r̶e̶v̶i̶e̶w̶S̶t̶a̶c̶k̶ ̶C̶r̶a̶f̶t̶ ̶A̶c̶t̶i̶o̶n̶.̶ ̶T̶h̶i̶s̶ ̶m̶i̶g̶h̶t̶ ̶i̶n̶d̶i̶c̶a̶t̶e̶ ̶a̶n̶ ̶e̶x̶p̶l̶o̶i̶t̶ ̶b̶y̶ ̶2̶5̶7̶a̶6̶6̶1̶7̶-̶c̶8̶c̶1̶-̶3̶e̶e̶4̶-̶a̶8̶5̶6̶-̶b̶9̶9̶b̶7̶7̶6̶f̶f̶0̶7̶9̶*̶*̶
edit: _(The problem with mechanism that did not allow construction when the tab did not move by itself was solved in the latest version of Miapi: [v1.1.2-1.20.1, but the tabs sometimes move on their own when you grab any other type of material, it seems that it was not the fault of the Mekanism.]_

**To Reproduce**

1. Choose a module of a tool on the workbench.
2. Select material to improve it.
3. The tab goes back.

**Expected behavior**

Place the materials in the module tab?

**Screenshots**

![image](https://github.com/Truly-Modular/Modular-Item-API/assets/126206944/aa47c6ae-0e5a-4e0a-b3ec-70264719ebc9)
![image](https://github.com/Truly-Modular/Modular-Item-API/assets/126206944/cf95e2ca-9d93-44a9-9baa-4a864a303644)

**Minecraft and Mod Version:**
Minecraft:
Forge/Fabric: neoforge-1.20.1-47.1.81
Miapi: v1.0.12-1.20.1
other:

**Comment by Smartin-b** [comment]
btw, is this fixed in teh latest release or still present?

**Comment by DankestO** [comment]
There are times when I enter the server it doesn't happen during the entire game, but the next day it may happen. it's strange.

Today while playing in singleplayer testing the armory module it happened to me quite a few times that the tab went back.

**Comment by DankestO** [comment]
When creating the issue I explained several things, they were all fixed in the last update except the tabs thing, I forgot to mention it.

**Comment by Smartin-b** [comment]
im having a trouble recreating this issue, did you notice some kind of pattern or anything?

**Comment by DankestO** [comment]
I don't know what causes it completely, I'm sorry, for now I have solved it by entering a world in singleplayer and later returning to multiplayer.
Maybe it's the fault of some performance mod that I have on the server, it wouldn't be the first time.

I've done some mod removal tests, but since this issue triggers pretty much randomly, it's kind of difficult if I'm on the right track or not.
I'll leave my current server mod list here, in case someone has the same problem as me and by chance we have a mod in common.

![image](https://github.com/Truly-Modular/Modular-Item-API/assets/126206944/0471efa1-02c1-431b-bbf2-01ef4a4f9d9c)

**Comment by Smartin-b** [comment]
oh, this could be caused by Domum Ornamentum reacting to the creation of stuff it detects as Weapon, adjusting the Item, causing Truly Modular to think you replaced the item in the slot causing it to refresh the gui

**Comment by Smartin-b** [comment]
we do call a check to see if the item is the same as before, but if Domum Ornamentum adds something to the item that might be bypassed... ill have a closer look soon

**Comment by Smartin-b** [comment]
could still not replicate, i tried in Singleplayer and dedicated server with Domum Ornamentum installed

**Comment by DankestO** [comment]
I tried putting on domum again, and I couldn't replicate it either, I feel so trolled by my own circumstance right now... xD

It only happens to me when I log into my dedicated server, I spent a while trying to remove all the performance mods on the server and testing if it was one of those. No luck, it didn't solve anything.

It also turns out that the day before the fix I had, I had also removed a couple of mods, but there was no improvement until I removed domum.
why? I'm so confused xD

I'm thinking that maybe it's not a problem with a mod itself, but with the amount of loading time? or something like that idk
I mean, on servers, as you explained to me the other day, you had to reload it twice for the mod materials to load properly, right?
I want to assume that it only retracts once and not indefinitely like before due to having fewer mods that load more or something like that?

After all, as I mentioned before, this problem is solved by loading a world in singleplayer, and after that, when entering the server this problem magically no longer occurs. Would it be like reloading the client? (?) It's something similar to what happens with dedicated servers, right?
Maybe I've already gone crazy (?)

**Comment by Smartin-b** [comment]
ill leave the issue open for now, maybe you or somebody else can figure something out...

**Comment by DankestO** [comment]
It's been a while since I've been here (?
I think I found the incompatibility, I think it's Passive Skill Tree, I saw that it does the same blink that I explained but this time in another mod.
https://github.com/Daripher/Passive-Skill-Tree/issues/178

I've been testing it for a while without PST, and I don't seem to have that problem.
Since in this case it would not be Miapi's problem, I am going to close this issue.
